routing.rst

```diff
@@ -843,87 +922,941 @@ You can also use special attributes to configure them (except ``_fragment``):
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Route parameters can contain any values except the ``/`` slash character,
+because that's the character used to separate the different parts of the URLs.
+For example, if the ``token`` value in the ``/share/{token}`` route contains a
+``/`` character, this route won't match.
+
+A possible solution is to change the parameter requirements to be more permissive:
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        use Symfony\Component\Routing\Annotation\Route;
+
+        class DefaultController
+        {
+            /**
+             * @Route("/share/{token}", name="share", requirements={"token"=".+"})
+             */
+            public function share($token)
+            {
+                // ...
+            }
+        }
+
+    .. code-block:: yaml
+
+        # config/routes.yaml
+        share:
+            path:       /share/{token}
+            controller: App\Controller\DefaultController::share
+            requirements:
+                token: .+
+
+    .. code-block:: xml
+
+        <!-- config/routes.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                https://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <route id="share" path="/share/{token}" controller="App\Controller\DefaultController::share">
+                <requirement key="token">.+</requirement>
+            </route>
+        </routes>
+
+    .. code-block:: php
+
+        // config/routes.php
+        use App\Controller\DefaultController;
+        use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+        return function (RoutingConfigurator $routes) {
+            $routes->add('share', '/share/{token}')
+                ->controller([DefaultController::class, 'share'])
+                ->requirements([
+                    'token' => '.+',
+                ])
+            ;
+        };
+
+.. note::
+
+    If the route defines several placeholders and you apply this permissive
+    regular expression to all of them, the results won't be the expected. For
+    example, if the route definition is ``/share/{path}/{token}`` and both
+    ``path`` and ``token`` accept ``/``, then ``path`` will contain its contents
+    and the token, and ``token`` will be empty.
+
+.. note::
+
+    If the route includes the special ``{_format}`` parameter, you shouldn't
+    use the ``.+`` requirement for the parameters that allow slashes. For example,
+    if the pattern is ``/share/{token}.{_format}`` and ``{token}`` allows any
+    character, the ``/share/foo/bar.json`` URL will consider ``foo/bar.json``
+    as the token and the format will be empty. This can be solved by replacing
+    the ``.+`` requirement by ``[^.]+`` to allow any character except dots.
```

Comment:
Won't the lazy regex quantifier (`.+?`) help here? If not, I think it's good to add it explicitly

Reply:
Not sure. I used the same expression which was in the previous doc. Changing these little things inside a huge refactorization is always scary. So, I prefer to leave this.